### PR TITLE
Hop contract updates

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- fixed bug of transferring native token

### CODE REVIEW

File: contracts/polygon/connectors/hop/events.sol [hunk 1]
```diff
@@ -4,6 +4,7 @@ pragma solidity ^0.7.0;
 contract Events {
 	event LogBridge(
 		address token,
+		bool isWrapped,
```
Comment (thrilok209) [inline contracts/polygon/connectors/hop/events.sol]: @pradyuman-verma Is this required in event?

File: contracts/polygon/connectors/hop/helpers.sol [hunk 2]
```diff
@@ -30,11 +31,27 @@ contract Helpers is DSMath, Basic {
...
+				params.amount,
+				params.bonderFee,
+				params.sourceAmountOutMin,
+				params.sourceDeadline,
+				params.destinationAmountOutMin,
+				params.destinationDeadline
+			);
+
+			return;
+		}
```
Comment (thrilok209) [inline contracts/polygon/connectors/hop/helpers.sol]:
```suggestion
		uint256 nativeTokenAmt = isNative ? params.amount : 0;
```

File: contracts/polygon/connectors/hop/helpers.sol [hunk 3]
```diff
@@ -30,11 +31,27 @@ contract Helpers is DSMath, Basic {
...
+
 		TokenInterface tokenContract = TokenInterface(params.token);
 		approve(tokenContract, params.router, params.amount);
```
Comment (thrilok209) [inline contracts/polygon/connectors/hop/helpers.sol]:
```suggestion
		router.swapAndSend{ value: nativeTokenAmt }(
```

File: contracts/polygon/connectors/hop/helpers.sol [hunk 4]
```diff
@@ -30,11 +31,27 @@ contract Helpers is DSMath, Basic {
 		uint256 sourceDeadline;
 		uint256 destinationAmountOutMin;
 		uint256 destinationDeadline;
+		bool isWrapped;
```
Comment (thrilok209) [inline contracts/polygon/connectors/hop/helpers.sol]: @pradyuman-verma Why send extra param? Instead, we can check this on smart contract only, right?

File: contracts/polygon/connectors/hop/events.sol (hunk 1 above)
Comment (pradyuman-verma) [reply contracts/polygon/connectors/hop/events.sol]: Not necessarily.

File: contracts/polygon/connectors/hop/helpers.sol (hunk 4 above)
Comment (pradyuman-verma) [reply contracts/polygon/connectors/hop/helpers.sol]: yes, we can check this on smart contract itself.

File: contracts/polygon/connectors/hop/main.sol [hunk 5]
```diff
@@ -37,24 +37,26 @@ abstract contract Resolver is Helpers {
...
 
-		bool isMatic = params.token == maticAddr;
-		params.token = params.token == maticAddr ? wmaticAddr : params.token;
+		if (params.token == wmaticAddr) {
+			convertWmaticToMatic(true, tokenContract, params.amount);
+			params.token = maticAddr;
+		}
 
-		TokenInterface tokenContract = TokenInterface(params.token);
+		bool isNative = params.token == maticAddr;
```
Comment (thrilok209) [inline contracts/polygon/connectors/hop/main.sol:47]: @pradyuman-verma We need to check if the token is wnative, then we need to unwrap it.

Comment (pradyuman-verma) [reply contracts/polygon/connectors/hop/main.sol:47]: Doing that part in line 42-45

**Review by thrilok209 — DISMISSED** [review]
Looks good! Let's deploy it!

**Comment by pradyuman-verma** [comment]
Name: `HOP-A`

### Deployed on polygon

Addr: [0x2a84370fE9DDFed08b076273Cd32cD39a49c6B01](https://polygonscan.com/address/0x2a84370fE9DDFed08b076273Cd32cD39a49c6B01)
**tx** [here](https://polygonscan.com/tx/0xcd8373e098455db34b6231e8c791142d7a13aa1c908e8e390740f09770d5f80e)

### Deployed on Arbitrum
Addr: [0x878c410028e3830f1fe03c428ff95012111ae1f1](https://arbiscan.io/address/0x878c410028e3830f1fe03c428ff95012111ae1f1#code)
tx: [here](https://arbiscan.io/tx/0xa2890622d3cf66719c4dc2c5fd1116e3cbdf52c3ef98b7e97822bc1724ae2584)

### Deployed on Optimism
Addr: [0xae7cd06693191c1f5f1477b9e9a567732d715f21](https://optimistic.etherscan.io/address/0xae7cd06693191c1f5f1477b9e9a567732d715f21)
tx: [here](https://optimistic.etherscan.io/tx/0x1a435696f8c6e45f7b04eec9d48b475ea0406811c85a0e4922deacfa84f239aa)

**Comment by pradyuman-verma** [comment]
Arb testcases: 
image

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**